app.js

```diff
@@ -36,143 +40,89 @@ function cookiesSoldThisHour(customers, avgCookies) {
-      var sales = cookiesSoldThisHour(customers, this.avgCookies);
-      this.totalSales += sales;
-      console.log(this.totalSales);
-      // console.log(hoursOpen[iHours] + ': ' + sales + ' cookies.');
-      createElement('li', 'class', 'hourly-sales', hoursOpen[iHours] + ': ' + sales + ' cookies.', elements[iStores]);
-      this.hourlySales.push(sales);
-    }
-  }
-};
+// Capitol Hill Store
+var hillStore = NewStore('Capitol Hill Store', 20, 38, 2.3, 'hill-store');
 
 // Alki Store
-var alkiStore = {
-  minCustomers: 2,
-  maxCustomers: 16,
-  avgCookies: 4.6,
-  totalSales: 0,
-  hourlySales: [],
-  cookieSales: function() {
-    for (var iHours = 0; iHours < hoursOpen.length; iHours++) {
-      var customers = customersThisHour(this.maxCustomers, this.minCustomers);
-      var sales = cookiesSoldThisHour(customers, this.avgCookies);
-      this.totalSales += sales;
-      console.log(this.totalSales);
-      // console.log(hoursOpen[iHours] + ': ' + sales + ' cookies.');
-      createElement('li', 'class', 'hourly-sales', hoursOpen[iHours] + ': ' + sales + ' cookies.', elements[iStores]);
-      this.hourlySales.push(sales);
-    }
-  }
-};
+var alkiStore = NewStore('Alki Store', 2, 16, 4.6, 'alki-store');
 
 //stores output
 var stores = [pikeStore, airportStore, centerStore, hillStore, alkiStore];
-for (var iStores = 0; iStores < stores.length; iStores++) {
-  stores[iStores].cookieSales();
-  createElement('li', 'class', 'store-total', 'Total: ' + stores[iStores].totalSales + ' cookies', elements[iStores]);
-}
 // populate and display the cookies data
-/*
+
+//produce headder
+function makeTableHead() {
+  var headParent = createElement('thead', 'id', 'table-head', 'Hours', tableEl);
+  for (var iHead = 0; iHead < hoursOpen.length; iHead++) {
+    createElement('th', 'class', 'time-indicator', hoursOpen[iHead], headParent);
+  }
+  createElement('th', 'id', 'totals', 'Total', headParent);
+}
+
+//produce footer
+function makeTableFoot() {
+  var footParent = createElement('tfoot', 'id', 'table-foot', 'Totals', tableEl);
+  var grandTotal = 0;
+  for (var iFoot = 0; iFoot < hoursOpen.length; iFoot++) {
+    var hourlyTotal = 0;
+    for (var iSubtotal = 0; iSubtotal < stores.length; iSubtotal++) {
+      hourlyTotal += stores[iSubtotal].hourlySales[iFoot];
+      grandTotal += stores[iSubtotal].hourlySales[iFoot];
+    }
+    createElement('td', 'class', 'hourly-totals', hourlyTotal, footParent);
+  }
+  createElement('td', 'id', 'grand-total', grandTotal, footParent);
+}
+
+makeTableHead();
+
 for (var iStores = 0; iStores < stores.length; iStores++) {
   console.log(stores[iStores]);
-  stores[iStores].hourlySales = cookiesSales(stores[iStores].maxCustomers, stores[iStores].minCustomers, stores[iStores].avgCookies);
-  createElement('li', 'class', 'store-total', 'Total: ' + stores[iStores].totalSales + ' cookies', elements[iStores]);
+  var rowParent = createElement('tr', 'id', stores[iStores].tag, stores[iStores].name, tableEl);
```

Comment:
In the lab instructions, there was a technical requirement to make a render() method for each store to handle creating and appending a store's row to the table.